internal/image/image.go

```diff
@@ -0,0 +1,65 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package image
+
+import (
+	"fmt"
+	"strings"
+
+	submariner "github.com/submariner-io/submariner-operator/api/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/pkg/images"
+	"github.com/submariner-io/submariner-operator/pkg/names"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/utils"
+)
+
+func Operator(imageVersion, repo string, imageOverrideArr []string) string {
+	if imageVersion == "" {
+		imageVersion = submariner.DefaultSubmarinerOperatorVersion
+	}
+
+	if repo == "" {
+		repo = submariner.DefaultRepo
+	}
+
+	return images.GetImagePath(repo, imageVersion, names.OperatorImage, names.OperatorComponent, GetOverrides(imageOverrideArr))
+}
+
+func GetOverrides(imageOverrideArr []string) map[string]string {
+	if len(imageOverrideArr) > 0 {
+		imageOverrides := make(map[string]string)
+		for _, s := range imageOverrideArr {
+			key := strings.Split(s, "=")[0]
+			if invalidImageName(key) {
+				utils.ExitWithErrorMsg(fmt.Sprintf("Invalid image name %s provided. Please choose from %q", key, names.ValidImageNames))
```

Comment:
This gets called from functions which are part of the library (`Broker()`), so it can’t output by itself or exit by itself. All it can do is return an error.

Reply:
Done